### ISSUE TITLE
OpenSSL command

### ISSUE THREAD

**Issue by macOneOone** [holzschu/a-Shell-commands#54]
@holzschu,

Is there any alternative for openSSL into a-Shell-commands?

Since we’ve ssh-keygen, it would be a great idea to have openSSL as built in package.

**Comment by holzschu** [comment]
What are the commands you're looking for? When I look at the source code for openSSL, I see only the library, no commands. I'm probably looking in the wrong place, but I need help.

**Comment by moon-cacke** [comment]
I need OpenSSL as well for certificate management and revocation status check... iSH app have it but it's not compatible with shortcuts... I would rather prefer having that package on a-shell... please consider adding it :)

**Comment by moon-cacke** [comment]
Or support a working pyopenssl package... thanks a lot :)

**Comment by robughblah** [comment]
﻿this might be a cool party trick for you. ash supports shortcuts and ssh client. ish can be configured as an ssh server. [~/Documents]$ uname -aDarwin localhost 24.2.0 Darwin Kernel Version 24.2.0: Thu Nov 14 22:52:09 PST 2024; root:xnu-11215.62.3~1/RELEASE_ARM64_T8020 iPad8,11[~/Documents]$ ssh -p 22000 ***@***.***@localhost's password: Welcome to Alpine!You can install packages with: apk add <package>You may change this message by editing /etc/motd.localhost:~# localhost:~# uname -aLinux localhost 4.20.69-ish SUPER AWESOME May 20 2023 23:41:32 i686 LinuxRobOn Jan 11, 2025, at 11:29 PM, moon-cacke ***@***.***> wrote:﻿
I need OpenSSL as well for certificate management and revocation status check... iSH app have it but it's not compatible with shortcuts... I would rather prefer having that package on a-shell... please consider adding it :)

**Comment by moon-cacke** [comment]
> this might be a cool party trick for you. ash supports shortcuts and ssh client. ish can be configured as an ssh server. [~/Documents]$ uname -aDarwin localhost 24.2.0 Darwin Kernel Version 24.2.0: Thu Nov 14 22:52:09 PST 2024; root:xnu-11215.62.3~1/RELEASE_ARM64_T8020 iPad8,11[~/Documents]$ ssh -p 22000 ***@***.***@localhost's password: Welcome to Alpine!You can install packages with: apk add <package>You may change this message by editing /etc/motd.localhost:~# localhost:~# uname -aLinux localhost 4.20.69-ish SUPER AWESOME May 20 2023 23:41:32 i686 LinuxRobOn Jan 11, 2025, at 11:29 PM, moon-cacke ***@***.***> wrote:﻿ I need OpenSSL as well for certificate management and revocation status check... iSH app have it but it's not compatible with shortcuts... I would rather prefer having that package on a-shell... please consider adding it :) —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

Yeah that's a workaround but ish server does remain on the background its not reliable... 

For me currently I did wrote a shortcut that uses Libterm, Python, Curl, Google's DNS and Apple's OSCP server to check the status of my target cert without anything running in the backgroup, via an ocsp-request.der file generated on iSH's Openssl... relatively complicated as it imply reading der's answer on a hex format but its works... took me sometimes to figure out that shit... I will post a detailed article on it on my blog when I have the time and link that here...
